node/tests/reconstruct_test.go

```diff
@@ -0,0 +1,120 @@
+//go:build !race
+
+package tests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/celestiaorg/celestia-node/ipld"
+
+	"github.com/celestiaorg/celestia-node/service/share"
+
+	"github.com/celestiaorg/celestia-node/node"
+	"github.com/celestiaorg/celestia-node/node/tests/swamp"
+)
+
+func TestFullReconstructFromBridge(t *testing.T) {
+	const (
+		blocks = 20
+		bsize  = 16
+		btime  = time.Millisecond * 100
+	)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	t.Cleanup(cancel)
+	sw := swamp.NewSwamp(t, swamp.WithBlockTime(btime))
+	go sw.FillBlocks(ctx, t, bsize, blocks)
+
+	bridge := sw.NewBridgeNode()
+	err := bridge.Start(ctx)
+	require.NoError(t, err)
+
+	full := sw.NewFullNode(node.WithTrustedPeers(getMultiAddr(t, bridge.Host)))
+	err = full.Start(ctx)
+	require.NoError(t, err)
+
+	errg, bctx := errgroup.WithContext(ctx)
+	for i := 1; i <= blocks+1; i++ {
+		i := i
+		errg.Go(func() error {
+			h, err := full.HeaderServ.GetByHeight(bctx, uint64(i))
+			if err != nil {
+				return err
+			}
+
+			return full.ShareServ.SharesAvailable(bctx, h.DAH)
+		})
+	}
+
+	err = errg.Wait()
+	require.NoError(t, err)
+}
+
```

Comment:
```suggestion

/*
Test-Case: Full Node reconstructs blocks only from Light Nodes
Pre-Reqs: 
- First 20 blocks have a block size of 16
- Blocktime is 100 ms
Steps:
1. Create a Bridge Node(BN)
2. Start a BN
3. Create 69 Light Nodes(LNs) with BN as a trusted peer
4. Start 69 LNs
5. Create a Full Node(FN) with 69 LNs as trusted peers
6. Unlink FN connection to BN
7. Start a FN
8. Check that a FN can retrieve shares from 1 to 20 blocks
*/
```

Reply:
Line 69 with a mention of 69
![image](https://user-images.githubusercontent.com/1822365/176467836-d624ccb1-1d5d-4f6c-a03c-611d44db3685.png)